### PR TITLE
Correct pull of data from MultiMapDataStore

### PR DESCRIPTION
**Pull request by menion**
Recent implementations of the `read` operations in the `MultiMapDataStore` are based on the checking of the upper-left tile only. 

This results in missing data in case of for example low zoom level and multiple maps covered by area defined by "upperLeft" and "lowerRight" tiles.

On the attached screen is an example of "readLabels" results, which should cover the whole screen, but the results are only data from the first DE map.

image

### CODE REVIEW

**Comment by devemux86** [comment]
@menion thanks!

This seems right, although I cannot reproduce the report with e.g. Mapsforge desktop samples app.
All this time multiple maps seem to render correctly.

How we can trigger this using the Mapsforge samples?

File: mapsforge-map-reader/src/main/java/org/mapsforge/map/reader/MapFile.java [hunk 1]
```diff
@@ -1072,6 +1072,11 @@ public boolean supportsTile(Tile tile) {
                 && (tile.zoomLevel >= this.zoomLevelMin && tile.zoomLevel <= this.zoomLevelMax);
     }
 
+    @Override
+    public boolean supportsArea(BoundingBox boundingBox) {
+        return boundingBox.intersects(getMapFileInfo().boundingBox);
```
Comment (devemux86) [inline mapsforge-map-reader/src/main/java/org/mapsforge/map/reader/MapFile.java:1077]: This should also contain the zoom level check like `supportsTile`.

**Comment by devemux86** [comment]
@menion 

I understand, please see my above 2 comments on the code.

File: mapsforge-map/src/main/java/org/mapsforge/map/datastore/MultiMapDataStore.java [hunk 2]
```diff
@@ -162,7 +162,7 @@ public MapReadResult readLabels(Tile upperLeft, Tile lowerRight) {
         switch (this.dataPolicy) {
             case RETURN_FIRST:
                 for (MapDataStore mdb : mapDatabases) {
-                    if (mdb.supportsTile(upperLeft)) {
+                    if (mdb.supportsArea(upperLeft.getBoundingBox().extendBoundingBox(lowerRight.getBoundingBox()))) {
```
[the remark this answers is not in the document]
Comment (menion) [reply mapsforge-map/src/main/java/org/mapsforge/map/datastore/MultiMapDataStore.java:165]: Could you be more precise? `Tile.getBoundingBox()` value is cached, so I do not expect and performance negative effect. Anyway if there is an option to do it faster ...

File: mapsforge-map-reader/src/main/java/org/mapsforge/map/reader/MapFile.java (hunk 1 above)
Comment (menion) [reply mapsforge-map-reader/src/main/java/org/mapsforge/map/reader/MapFile.java:1077]: My initial ideal was that zoom level should not play the role here, but as `MapFile` has it's defined limits, you are of course correct that it has to be there as well. I'll improve it.

**Pull request merged**